### DISCUSSION TITLE
How to override a TypeConverter?

### DISCUSSION THREAD

**Discussion by de-construct-mark** [phpro/soap-client#394] [General]
Hey guys, I love this tool, makes working with SOAP a breeze.... I have a question regarding the TypeConverters, specifically about the included DateTimeTypeConverter... basically, I just need to change the date format that it outputs. Is there a way to achieve this without changing the file inside the package in vendor folder?
Any help would be much appreciated.
Thanks, Mark

**Comment by de-construct-mark** [comment]
Actually ignore, i found a way via composers autoload:

"exclude-from-classmap" : ["vendor/phpro/src/soap-client/Soap/TypeConverter/DateTimeTypeConverter.php"] 
"files" : ["overrides/DateTimeTypeConverter.php"]

Seems to be working without any issues.

**Reply by veewee** [reply]
Hello @de-construct-mark,

We moved these classes into a separate repository for version 2:
https://github.com/php-soap/ext-soap-engine/tree/main/src/Configuration/TypeConverter

Currently it is not possible to change the format. We could change the implementation so that you can provide a format through the constructor. This might make the class easier to use. 

So for now, the suggested approach would be to create your own type converter. You can use that by changing the typemap inside the soap options class.
